pkg/controller/util.go

```diff
@@ -326,3 +329,297 @@ func addToMap(m1, m2 map[string]string) map[string]string {
 	}
 	return m1
 }
+
+// returns the CloneRequest string which contains the pvc name (and namespace) from which we want to clone the image.
+func getCloneRequestPVC(pvc *v1.PersistentVolumeClaim) (string, error) {
+	cr, found := pvc.Annotations[AnnCloneRequest]
+	if !found || cr == "" {
+		verb := "empty"
+		if !found {
+			verb = "missing"
+		}
+		return cr, errors.Errorf("annotation %q in pvc \"%s/%s\" is %s\n", AnnCloneRequest, pvc.Namespace, pvc.Name, verb)
+	}
+	return cr, nil
+}
+
+func CreateCloneSourcePod(client kubernetes.Interface, image, verbose, pullPolicy, cr, secretName string, pvc *v1.PersistentVolumeClaim) (*v1.Pod, error) {
+	strArr := strings.Split(cr, "/")
+    var ns string = ""
```

Comment:
nit: indent
Good idea to run all new code through gofmt, eg `gofmt -w -s list-of-files`